### PR TITLE
VoiceClient updates

### PR DESCRIPTION
**Pull request by key2peace**
- set readPointer to 0 when closing buffer
- stop writing to the buffer if audio is going to be stopped anyways
- removed $count in playDCAStream as it served no purpose
- when stopping audio, also end/close the buffer
  also send 5 frames of silence as per https://discord.com/developers/docs/topics/voice-connections#voice-data-interpolation
- remove obsolete $this->setSpeaking(false); since $this->reset(); already does that

### CODE REVIEW

File: src/Discord/Voice/VoiceClient.php [hunk 1]
```diff
@@ -1492,4 +1522,16 @@ public function getChannel(): Channel
     {
         return $this->channel;
     }
+
+    /**
+     * Insert 5 frames of silence.
+     *
```
Comment (SQKo) [inline src/Discord/Voice/VoiceClient.php:1528]:
```suggestion
```

File: src/Discord/Voice/VoiceClient.php [hunk 2]
```diff
@@ -1492,4 +1522,16 @@ public function getChannel(): Channel
...
+     */
+    private function insertSilence(): void
+    {
+        // https://discord.com/developers/docs/topics/voice-connections#voice-data-interpolation
+        while ($silenceRemaining--) {
```
Comment (SQKo) [inline src/Discord/Voice/VoiceClient.php:1533]:
```suggestion
        while ($this->silenceRemaining--) {
```

**Review by SQKo — CHANGES_REQUESTED** [review]

File: src/Discord/Voice/VoiceClient.php [hunk 3]
```diff
@@ -1100,18 +1119,27 @@ public function unpause(): void
 
     /**
      * Stops the current sound.
+     *
+     * @return ExtendedPromiseInterface
+     * @throws \Exception
      */
-    public function stop(): void
+    public function stop(): ExtendedPromiseInterface
```
Comment (SQKo) [inline src/Discord/Voice/VoiceClient.php:1126]: Are you sure this wont allow the code being repeated?
```php
$vc->stop();
$vc->stop();
```

Comment (key2peace) [reply src/Discord/Voice/VoiceClient.php:1126]: it wouldn't matter if you repeat it, the second would return the exception that the audio is no longer playing

File: src/Discord/Voice/VoiceClient.php [hunk 4]
```diff
@@ -1119,27 +1119,18 @@ public function unpause(): void
...
+    public function stop(): void
     {
-        $deferred = new Deferred();
-
         if (! $this->speaking) {
-            $deferred->reject(new \Exception('Audio must be playing to stop it.'));
+            throw new \Exception('Audio must be playing to stop it.'));
 
-            return $deferred->promise();
+            return;
```
Comment (davidcole1340) [inline src/Discord/Voice/VoiceClient.php:1128]:
```suggestion
```

File: src/Discord/Voice/VoiceClient.php [hunk 5]
```diff
@@ -1103,15 +1122,15 @@ public function unpause(): void
...
     public function stop(): void
     {
-        if ($this->stopAudio) {
-            throw new \Exception('Audio is already being stopped.');
-        }
-
         if (! $this->speaking) {
             throw new \Exception('Audio must be playing to stop it.');
+
+            return;
```
Comment (davidcole1340) [inline src/Discord/Voice/VoiceClient.php:1128]:
```suggestion
```

**Review by davidcole1340 — APPROVED** [review]

**Pull request merged**